### PR TITLE
Adding types to ConfigInterface for callbacks

### PR DESCRIPTION
**Pull request by aequasi**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export interface ConfigInterface {
+export interface ConfigInterface<Data = any, Error = any> {
```
Comment (aequasi) [inline src/types.ts]: Wasnt sure if `Error` should default to `any` or `Error`

Comment (aequasi) [reply src/types.ts]: Added defaults to avoid a BC break

Comment (quietshu) [reply src/types.ts]: Yeah, I think it's good to leave the possibility of throwing anything. There might be some interesting use cases.

**Comment by quietshu** [comment]
You may need to run `yarn lint:fix` to fix the prettier error 🙏

**Comment by aequasi** [comment]
Fixed! Thanks

**Pull request merged**